src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/NodeFactory.cs

```diff
@@ -106,7 +113,14 @@ private void CreateNodeCaches()
             {
                 if (_compilationModuleGroup.ContainsType(type))
                 {
-                    return new EETypeNode(type, true);
+                    if (type.IsGenericDefinition)
```

Comment:
Do we still need this?

(Also, could you not squash the next time so that I can review the diff?)